### ISSUE TITLE
hparams with separate train and validation summary writers

### ISSUE THREAD

**Issue by saravanabalagi** [tensorflow/tensorboard#3675]
I have a model that I train with `hp.KerasCallback` and `TensorBoard` callback as explained in the [example](https://www.tensorflow.org/tensorboard/hyperparameter_tuning_with_hparams#2_adapt_tensorflow_runs_to_log_hyperparameters_and_metrics): 
```
model.fit(
    ...,
    callbacks=[
        tf.keras.callbacks.TensorBoard(logdir),  # log metrics
        hp.KerasCallback(logdir_hp, hparams),  # log hparams
    ],
)
```
This successfully prints all my input (hyper)parameters (in hparams dict) and
- train metrics, if `logdir_hp` is `/train` (or a summary writer to train dir)
- validation metrics, if `logdir_hp` is `/validation` (or a summary writer to validation dir)

How do I log the same metric but in both runs (train and validation) for better analysis?

![image](https://user-images.githubusercontent.com/8567893/83083908-4d83de80-a07f-11ea-95e5-1cc38df8f7d6.png)

The above result was obtained using estimator api which automatically appends the run name based on the folder (train on current folder, so no prefix, for validation it's `eval` and I had an additional evaluation every x steps saved in `global_eval`), is it possible to achieve the same in keras model.fit()?

**Comment by saravanabalagi** [comment]
Will something like a prefix option to callback feasible?

```
model.fit(
    ...,
    callbacks=[
        tf.keras.callbacks.TensorBoard(logdir),
        hp.KerasCallback(logdir_train, hparams, prefix=""), 
        hp.KerasCallback(logdir_validation, hparams, prefix="val_"), 
    ],
)
```
and then somehow map any metric logged under `logdir_validation` to carry display name `val_metric`.

**Comment by saravanabalagi** [comment]
I tried digging into the source and found that there hparams get logged in [on_train_begin](https://github.com/tensorflow/tensorboard/blob/3e20cdb240ffeeee258c7bd13f93089fe7786619/tensorboard/plugins/hparams/keras.py#L89) and raw protobuf gets written in [on_train_end](https://github.com/tensorflow/tensorboard/blob/3e20cdb240ffeeee258c7bd13f93089fe7786619/tensorboard/plugins/hparams/keras.py#L93). With my limited knowledge on pb, I didn't succeed in understanding serialized pb written as summary, or where to look in order to add a prefix and map back. Any help is appreciated. Thanks.

**Comment by bmd3k** [comment]
Hello! Thanks for reporting your issue. In general it is possible to log and differentiate between training and validation metrics in the hparams dashboard. In fact I made a copy and modified the [hparams example](https://www.tensorflow.org/tensorboard/hyperparameter_tuning_with_hparams#2_adapt_tensorflow_runs_to_log_hyperparameters_and_metrics) so that it uses Keras callbacks and validation and this was the result:


![image](https://user-images.githubusercontent.com/17152369/83258656-9f625c80-a184-11ea-9ace-ec8cf8e2dc5b.png)

Perhaps you could give us a small working example that reproduces your problem and we can make some suggestions or dig into it?

**Comment by saravanabalagi** [comment]
Hi @bmd3k , thanks for your reply. Just seeing the reply now, here's some working code on [colab](https://colab.research.google.com/drive/1aIWpRZll5j_8yG_T8wZVePHr1FwECayk?usp=sharing). You can toggle the checkbox `train_dir_on` to see that only one of train or validation metrics get's logged to tensorboard based on your selection. Let me know if you have any trouble accessing it.

**Comment by saravanabalagi** [comment]
Here are the screenshots showing the same:

`logdir_hp` pointing to `/train` i.e. `train_dir_on=True`:

![image](https://user-images.githubusercontent.com/8567893/84075508-307ad400-a9cc-11ea-8f4d-6396f5df6f68.png)

`logdir_hp` pointing to `/validation` i.e. `train_dir_on=False`:

![image](https://user-images.githubusercontent.com/8567893/84075650-67e98080-a9cc-11ea-91ec-9a5545b58800.png)
